tests/foreman/cli/test_syncplan.py

```diff
@@ -273,3 +314,102 @@ def test_verify_bugzilla_1261122(self):
         new_sync_plan = self._make_sync_plan()
         result = SyncPlan.info({'id': new_sync_plan['id']})
         self.assertIsNotNone(result.get('enabled'))
+
+    @tier2
+    def test_positive_synchronize_product(self):
+        """Create a sync plan with one product and sync it automatically.
+
+        @Assert: Product is synchronized successfully.
+
+        @Feature: SyncPlan
+        """
+        org = make_org()
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        product = make_product({'organization-id': org['id']})
+        repo = make_repository({'product-id': product['id']})
+        self.validate_repo_content(
+            repo, ['errata', 'packages'], after_sync=False)
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        self.validate_repo_content(
+            repo, ['errata', 'package-groups', 'packages'])
+
+    @tier2
+    def test_positive_synchronize_products(self):
+        """Create a sync plan with multiple products and sync them
+        automatically.
+
+        @Assert: Products are synchronized successfully.
+
+        @Feature: SyncPlan
+        """
+        org = make_org()
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        products = [
+            make_product({'organization-id': org['id']})
+            for _ in range(randint(3, 5))
+        ]
+        repos = [
+            make_repository({'product-id': product['id']})
+            for product in products
+            for _ in range(randint(2, 3))
+        ]
+        for repo in repos:
+            self.validate_repo_content(
+                repo, ['errata', 'packages'], after_sync=False)
+        for product in products:
+            Product.set_sync_plan({
+                'id': product['id'],
+                'sync-plan-id': sync_plan['id'],
+            })
+        for repo in repos:
+            self.validate_repo_content(
+                repo, ['errata', 'package-groups', 'packages'])
+
+    @tier2
+    def test_positive_synchronize_future_sync_date(self):
+        """Create a sync plan with sync date in future and sync one product
+        with it automatically.
+
+        @Assert: Product is synchronized successfully.
+
+        @Feature: SyncPlan
+        """
+        delay = 10 * 60  # delay for sync date in seconds
```

Comment:
For other reviewers - it won't overlap, as the first 2 checks will either pass without any delay (as repo is expected to have no content) or will fail the test in 5 minutes in case some content was found.